### PR TITLE
Add captains log analyzer

### LINKED ISSUE #82: Write analysis for concept exercise `captains-log`
Desired checks:
https://github.com/exercism/elixir/blob/9774308bb6b898df6ecbde64c5c6e48c651870ae/exercises/concept/captains-log/.meta/design.md#L28-L32

### PR DESCRIPTION
**Pull request by joseemds**
Waiting for https://github.com/exercism/website-copy/pull/2094 be merged to add more tests

closes #82

### CODE REVIEW

File: test/elixir_analyzer/test_suite/captains_log_test.exs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      def random_stardate() do
+        :rand.uniform() * 1000 + 41_000
+      end
+
+      def format_stardate(stardate) do
+        to_string(:io_lib.format("~.1f", [stardate]))
+      end
+    end
+  end
+end
```
Comment (angelikatyborska) [inline test/elixir_analyzer/test_suite/captains_log_test.exs:28]: Could you also add test cases that will check when the comments get triggered? At least one test for each analyzer comment. The code snippets used in the tests don't need to be full solutions, they can be partial solutions. Something like for this exercise: https://github.com/exercism/elixir-analyzer/blob/ca5fd50ff10ba572186690473d13b2befd9ccc41/test/elixir_analyzer/test_suite/bird_count_test.exs#L27-L165

Comment (angelikatyborska) [reply test/elixir_analyzer/test_suite/captains_log_test.exs:28]: Ah, sorry, I just noticed the PR description that says:
> Waiting for exercism/website-copy#2094 be merged to add more tests

There's no need to wait for the website-copy PR to add more tests. The tests for comments assert on the comment constants, not the actual content of the markdown files 🙂. Anyway, I will merge the PR now.

**Comment by angelikatyborska** [comment]
Looks good so far!

Btw. we added a small change to the tests recently that makes them depend on the git submodule. Here's a PR that adds this info to the README: https://github.com/exercism/elixir-analyzer/pull/182/files

This step will be necessary to reproduce this test failure that happens on CI:

```

  1) test example solution (ElixirAnalyzer.ExerciseTest.CaptainsLogTest)
     test/elixir_analyzer/test_suite/captains_log_test.exs:7
     Assertion with == failed
     code:  assert Enum.sort(comments) == Enum.sort(expected_comments)
     left:  ["elixir.solution.same_as_exemplar"]
     right: []
     stacktrace:
       (elixir 1.12.1) lib/enum.ex:1553: Enum."-map/2-lists^map/1-0-"/2
       test/elixir_analyzer/test_suite/captains_log_test.exs:7: (test)
```

Basically if a solution is 100% identical to the exemplar solution, the analyzer leaves this comment: https://github.com/exercism/elixir-analyzer/blob/ca5fd50ff10ba572186690473d13b2befd9ccc41/test/elixir_analyzer/test_suite/bird_count_test.exs#L6

**Comment by joseemds** [comment]
I couldn't think into many different wrong solutions, so i added just one test for each analyzer case. If you think that more tests will be needed please let me know!

**Review by angelikatyborska — APPROVED** [review]
One wrong snippet per comment is enough! I just did a small adjustments so that the wrong implementation would actually pass the exercise's test, just for the purpose of clarity (to prove that the analyzer check is necessary).

Thank you for your work!

**Pull request merged**